dmaws/github.py

```diff
@@ -0,0 +1,83 @@
+from datetime import datetime
+
+import requests
+
+
+def create_github_deployment(token, repo, ref, environment, payload=None):
+    return requests.post(
+        'https://api.github.com/repos/{}/deployments'.format(repo),
+        headers={'Authorization': 'token {}'.format(token)} if token else None,
+        json={
+            'ref': ref,
+            'auto_merge': False,
+            'required_contexts': [],
+            'environment': environment,
+            'payload': payload
+        }
+    )
+
+
+def set_github_deployment_status(token, repo, deployment_id, state, target_url=None):
+    return requests.post(
+        'https://api.github.com/repos/{}/deployments/{}/statuses'.format(repo, deployment_id),
+        headers={'Authorization': 'token {}'.format(token)} if token else None,
+        json={
+            'state': state,
+            'target_url': target_url,
+        }
+    )
+
+
+def get_github_deployments(token, repo, ref=None, environment=None):
+    return requests.get(
+        'https://api.github.com/repos/{}/deployments'.format(repo),
+        headers={'Authorization': 'token {}'.format(token)} if token else None,
+        params={'ref': ref, 'environment': environment}
+    )
+
+
+def publish_deployment(token, repo, ref, environment, build, created_at, ci_url, status=None,
+                       logger=None):
+    log = logger or (lambda *args, **kwargs: None)
+
+    ref_deployments = get_github_deployments(token, repo, ref, environment)
+    if ref_deployments.status_code == 200 and ref_deployments.json():
+        log('Deployment for {}@{} to {} already exists'.format(repo, ref, environment))
+        return True
+
+    created_at = created_at.isoformat()[:-7] + 'Z'
```

Comment:
Feels like we could do with a test around this setting the right value for `created_at`.